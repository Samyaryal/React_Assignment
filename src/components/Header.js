import React from 'react';
 
const Header = () => {
   return (
       <div  className="text-center mb-4">
           <h1 data-testid = "header">Todo List</h1>
       </div>
   );
};
 
export default Header;