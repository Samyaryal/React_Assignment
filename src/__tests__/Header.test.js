
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Header from '../components/Header';

//Individual Node Testing: testing with testId
it('Renders the header', () => {
  const { getByTestId} = render(<Header />);
  expect (getByTestId('header')).toBeTruthy()
});

//Snapshot testing
it('Renders the headers', () => {
  const { asFragment } = render(<Header text='Todo List' />);
  expect(asFragment()).toMatchSnapshot();
});