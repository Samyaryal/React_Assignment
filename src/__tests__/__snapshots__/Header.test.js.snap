// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the headers 1`] = `
<DocumentFragment>
  <div
    class="text-center mb-4"
  >
    <h1
      data-testid="header"
    >
      Todo List
    </h1>
  </div>
</DocumentFragment>
`;

exports[`renders the headers 1`] = `
<DocumentFragment>
  <div
    class="text-center mb-4"
  >
    <h1
      data-testid="header"
    >
      Todo List
    </h1>
  </div>
</DocumentFragment>
`;
